llvm/lib/Target/RISCV/RISCVISelDAGToDAG.cpp

```diff
@@ -1015,10 +1015,17 @@ void RISCVDAGToDAGISel::Select(SDNode *Node) {
       break;
     // If the mask has 32 trailing ones, use SRLIW.
     if (TrailingOnes == 32) {
-      SDNode *SRLIW =
-          CurDAG->getMachineNode(RISCV::SRLIW, DL, VT, N0->getOperand(0),
-                                 CurDAG->getTargetConstant(ShAmt, DL, VT));
-      ReplaceNode(Node, SRLIW);
+      if (Subtarget->is64Bit()) {
```

Comment:
Can we just select the opcode using `Subtarget->is64Bit()` instead of duplicating similar code?